src/contractor/contractor.cpp

```diff
@@ -299,18 +300,17 @@ parse_segment_lookup_from_csv_files(const std::vector<std::string> &segment_spee
         const auto file_id = idx + 1; // starts at one, zero means we assigned the weight
         const auto filename = segment_speed_filenames[idx];
 
-        std::ifstream segment_speed_file{filename, std::ios::binary};
-        if (!segment_speed_file)
-            throw util::exception{"Unable to open segment speed file " + filename};
-
+        storage::io::FileReader segment_speed_file_reader(std::string(filename),
+                                                          storage::io::FileReader::HasNoFingerprint);
```

Comment:
segment_speed_filenames is a ref to vector of strings; filename is segment_speed_filenames[idx];

why the extra string construction here?